core/esmf-aspect-model-aas-generator/src/main/java/org/eclipse/esmf/aspectmodel/aas/AasDataTypeMapper.java

```diff
@@ -0,0 +1,109 @@
+package org.eclipse.esmf.aspectmodel.aas;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.vocabulary.XSD;
+import org.eclipse.digitaltwin.aas4j.v3.model.DataTypeDefXSD;
+
+import java.util.Map;
+
+public class AasDataTypeMapper {
+
+    private AasDataTypeMapper() {
+    }
+
+    /**\
+     * Maps Aspect types to DataTypeDefXSD Schema types, with no explicit mapping defaulting to
+     * string
+     */
+    static final Map<Resource, DataTypeDefXSD> ASPECT_TYPE_TO_AAS_XSD_TYPE_MAP =
+        ImmutableMap.<Resource, DataTypeDefXSD> builder()
+            .put( XSD.anyURI, DataTypeDefXSD.ANY_URI )
+            .put( XSD.yearMonthDuration, DataTypeDefXSD.DURATION )
+            .put( XSD.xboolean, DataTypeDefXSD.BOOLEAN )
+            .put( XSD.xbyte, DataTypeDefXSD.BYTE )
+            .put( XSD.date, DataTypeDefXSD.DATE )
+            .put( XSD.dateTime, DataTypeDefXSD.DATE_TIME )
+            .put( XSD.dateTimeStamp, DataTypeDefXSD.DATE_TIME )
+            .put( XSD.dayTimeDuration, DataTypeDefXSD.DURATION )
+            .put( XSD.decimal, DataTypeDefXSD.DECIMAL )
+            .put( XSD.xdouble, DataTypeDefXSD.DOUBLE )
+            .put( XSD.duration, DataTypeDefXSD.DURATION )
+            .put( XSD.xfloat, DataTypeDefXSD.FLOAT )
+            .put( XSD.gMonth, DataTypeDefXSD.GMONTH )
+            .put( XSD.gMonthDay, DataTypeDefXSD.GMONTH_DAY )
+            .put( XSD.gYear, DataTypeDefXSD.GYEAR )
+            .put( XSD.gYearMonth, DataTypeDefXSD.GYEAR_MONTH )
+            .put( XSD.hexBinary, DataTypeDefXSD.HEX_BINARY )
+            .put( XSD.xint, DataTypeDefXSD.INT )
+            .put( XSD.integer, DataTypeDefXSD.INTEGER )
+            .put( XSD.xlong, DataTypeDefXSD.LONG )
+            .put( XSD.negativeInteger, DataTypeDefXSD.NEGATIVE_INTEGER )
+            .put( XSD.nonNegativeInteger, DataTypeDefXSD.NON_NEGATIVE_INTEGER )
+            .put( XSD.positiveInteger, DataTypeDefXSD.POSITIVE_INTEGER )
+            .put( XSD.xshort, DataTypeDefXSD.SHORT )
+            .put( XSD.normalizedString, DataTypeDefXSD.STRING )
+            .put( XSD.time, DataTypeDefXSD.TIME )
+            .put( XSD.unsignedByte, DataTypeDefXSD.UNSIGNED_BYTE )
+            .put( XSD.unsignedInt, DataTypeDefXSD.UNSIGNED_INT )
+            .put( XSD.unsignedLong, DataTypeDefXSD.UNSIGNED_LONG )
+            .put( XSD.unsignedShort, DataTypeDefXSD.UNSIGNED_SHORT )
+            .build();
+
+    /**\
+     * Maps DataTypeDefXSD Schema types to Aspect types, with no explicit mapping defaulting to
+     * normalizedString.
+     */
+    static final Map<DataTypeDefXSD, Resource> AAS_XSD_TYPE_TO_ASPECT_TYPE_MAP =
```

Comment:
Yes of course, but we have some mismatching of these types.
For example:
in Aspect we can see types: dateTime, dateTimeStamp. But in AASX just one type - DATE_TIME.
And I happened with few types. Of course, we can filter it. But I think that this way it is more clear and more understandable which types of their AASX correspond to the types of Asp net.

@ysrbo What do you think?

Thanks,
Best Regards!

Reply:
In the `AspectModelAASVisitor` a mapping from XSD to AAS types (IEC 61360) already exists. Please try to consolidate so that there is only one place where SAMM/XSD/AAS type mapping takes place.